src/commands/deployWorkspaceProject/getDefaultValues/DefaultResourcesNameStep.ts

```diff
@@ -0,0 +1,139 @@
+        const workspaceName: string = cleanWorkspaceName(nonNullValueAndProp(context.rootFolder, 'name'));
+        if (this.validateInput(workspaceName) !== undefined) {
+            return;
+        }
+
+        if (!await this.isWorkspaceNameAvailable(context, workspaceName)) {
+            return;
+        }
+
+        if (!context.resourceGroup || !context.managedEnvironment || !context.registry || !context.containerApp) {
+            ext.outputChannel.appendLog(localize('usingWorkspaceName', 'Using workspace name "{0}" as the default for remaining resource creation.', workspaceName));
+        }
+
+        !context.resourceGroup && (context.newResourceGroupName = workspaceName);
+        !context.managedEnvironment && (context.newManagedEnvironmentName = workspaceName);
+        !context.registry && (context.newRegistryName = workspaceName.replace(/[^a-z0-9]+/g, ''));
+        !context.containerApp && (context.newContainerAppName = workspaceName);
+        context.imageName = `${context.containerApp?.name || workspaceName}:latest`;
+    }
+
+    public shouldPrompt(context: DeployWorkspaceProjectContext): boolean {
+        return (!context.resourceGroup && !context.newResourceGroupName) ||
+            (!context.managedEnvironment && !context.newManagedEnvironmentName) ||
+            (!context.registry && !context.newRegistryName) ||
+            (!context.containerApp && !context.newContainerAppName);
+    }
+
+    private validateInput(name: string | undefined): string | undefined {
+        name ??= '';
+
+        // No symbols are allowed for ACR - we will strip out any offending characters from the base name, but still need to ensure this version has an appropriate length
+        const nameWithoutSymbols: string = name.replace(/[^a-z0-9]+/g, '');
+        if (!validateUtils.isValidLength(nameWithoutSymbols, 5, 20)) {
+            return localize('invalidLength', 'The alphanumeric portion of the name should total to at least 5 characters while not exceeding 20 characters.');
+        }
+
+        const symbols: string = '-';
+        if (!validateUtils.isLowerCaseAlphanumericWithSymbols(name, symbols, false /** canSymbolsRepeat */)) {
+            return validateUtils.getInvalidLowerCaseAlphanumericWithSymbolsMessage(symbols);
+        }
+
+        return undefined;
+    }
+
+    protected async validateNameAvailability(context: DeployWorkspaceProjectContext, name: string): Promise<string | undefined> {
+        return await window.withProgress({
+            location: ProgressLocation.Notification,
+            cancellable: false,
+            title: localize('verifyingAvailabilityTitle', 'Verifying resource name availability...')
+        }, async () => {
+            const resourceGroupAvailable: boolean = await ResourceGroupListStep.isNameAvailable(context, name);
+            const managedEnvironmentAvailable: boolean = await ManagedEnvironmentNameStep.isNameAvailable(context, name, name);
```

Comment:
I think that this could be optimized quite a bit. I'm a little concerned about performance since it's 4 async calls.

1. If the resourceGroupAvailable is true, then managedEnviornment and containerApp must be available as well.
  > Why? Because both of these resources only need to be unique per RG. If the RG name is available, we are able to infer that ME and CA will also be available since it's going to be a new RG.
2. If managedEnvironment is available, since story for containerApp. Similar to the logic above.
3. I assume the implementation right now is that we are not re-using an existing RG, right? If that's the case, then we can actually stop at `isResourceGroupNameAvailable` because, assuming we're creating a new one for them regardless, then points 1 and 2 are always true.
4. We will always have to check registry since it's agnostic to RG. I think it has to be globally unique.

Reply:
That makes sense, I agree with all of your points.  The only thing is for point 3, we do re-use resource groups in cases where we choose the most-used managed environment to house the container app.

Reply:
Ah, then in that case, I think 1 and 2 are still true, but disregard 3.

Reply:
Since we talked about removing the auto selection logic for managed environment, point 3 is now correct!

Reply:
So I know we went back and talked about removing the auto select on managed environments, so point 3 would technically be correct again since the prompt step would skip altogether if re-deploying and all resources are already there.  

However, I did decide to keep the logic generally similar (mainly because I'm lazy but also...) in case we ever decide we want to be able to handle partial resource availability.  It will be a little bit over-engineered but will be more re-usable/future-proof.  I didn't feel like going back and changing all the logic since it still technically works for this scenario.

Reply:
I'm not really sure how you'd ever have a case where you have a new environment but a invalid container name, but that's fine. I'm sure you have thought about it longer than me

Reply:
Nah, that was an oversight, thank you for catching that 😂